ext/mysqli/mysqli.c

```diff
@@ -772,6 +772,10 @@ PHP_MINIT_FUNCTION(mysqli)
 	REGISTER_LONG_CONSTANT("MYSQLI_TRANS_COR_RELEASE", TRANS_COR_RELEASE, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("MYSQLI_TRANS_COR_NO_RELEASE", TRANS_COR_NO_RELEASE, CONST_CS | CONST_PERSISTENT);
 
+#ifdef MARIADB_BASE_VERSION
+	REGISTER_LONG_CONSTANT("MYSQLI_IS_MARIADB", 1, CONST_CS | CONST_PERSISTENT);
```

Comment:
Hm, I wonder whether a) this should be a bool instead and b) it should always be defined with a true/false value, rather than conditionally defined. @kamil-tekiela Any thoughts?

Reply:
I'm not convinced we need such constant at all. What is the use case for this?

Reply:
well I used primarily for the test, otherwise can be useful when you do not own the server and wonder why so or so is not available.

Reply:
I think the usual way MariaDB is currently detected is checking whether the version is >= 100000.

Reply:
Ok, I am a little indifferent to it. It won't hurt to add it, but on the other hand, you should be able to get the same information using `mysqli_get_client_info`. Ideally, I would love to see the support for anything else than mysqlnd dropped. It just complicates everything unnecessarily. 

If we keep adding constants like this we will end up with `IS_MYSQL`, `IS_MARIADB`, `IS_MYSQLND`, etc.